homeassistant/components/samsungtv/bridge.py

```diff
@@ -0,0 +1,260 @@
+"""samsungctl and samsungtvws bridge classes."""
+from abc import ABC, abstractmethod
+
+from samsungctl import Remote
+from samsungctl.exceptions import AccessDenied, ConnectionClosed, UnhandledResponse
+from samsungtvws import SamsungTVWS
+from samsungtvws.exceptions import ConnectionFailure
+from websocket import WebSocketException
+
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_ID,
+    CONF_METHOD,
+    CONF_NAME,
+    CONF_PORT,
+    CONF_TIMEOUT,
+    CONF_TOKEN,
+)
+
+from .const import (
+    CONF_DESCRIPTION,
+    LOGGER,
+    METHOD_LEGACY,
+    RESULT_AUTH_MISSING,
+    RESULT_NOT_SUCCESSFUL,
+    RESULT_NOT_SUPPORTED,
+    RESULT_SUCCESS,
+    VALUE_CONF_ID,
+    VALUE_CONF_NAME,
+)
+
+
+class SamsungTVBridge(ABC):
+    """The Base Bridge abstract class."""
+
+    @staticmethod
+    def get_bridge(method, host, port=None, token=None):
+        """Get Bridge instance."""
+        if method == METHOD_LEGACY:
+            return SamsungTVLegacyBridge(method, host, port)
+        return SamsungTVWSBridge(method, host, port, token)
+
+    def __init__(self, method, host, port, token):
+        """Initialize Bridge."""
+        self.port = None
+        self.token = token
+        self.method = method
+        self.host = host
+        self.config = {
+            CONF_NAME: VALUE_CONF_NAME,
+            CONF_DESCRIPTION: VALUE_CONF_NAME,
+            CONF_ID: VALUE_CONF_ID,
```

Comment:
Make sure we don't create a config that doesn't match the subclass bridge specific config here. Only add the common items here.